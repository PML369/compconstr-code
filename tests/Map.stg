map = {} \n {f,xs} ->
        case xs {} of
            Nil {}      -> Nil {};
            Cons {y,ys} -> let fy = {f,y} \u {} -> f {y};
                               mfy = {f,ys} \u {} -> map {f,ys};
                               in Cons {fy,mfy};
            default     -> Nil {};
