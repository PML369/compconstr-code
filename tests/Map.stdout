Parsing tests/Map.stg...
<1:1>map = {} \n {f,xs} -> <2:9>case <2:14>xs {} of
                           <3:13>Nil -> <3:28>Nil {};
                           <4:13>Cons y ys -> <4:28>let
                                              <4:32>fy = {f,y} \u {} -> <4:52>f {<4:55>y};
                                              <5:32>mfy = {f,ys} \u {} -> <5:54>map {<5:59>f,<5:61>ys} in <6:35>Cons {<6:41>fy,<6:44>mfy};
                           <7:13>default -> <7:28>Nil {}
